kubernetes/helm/pinot/templates/minion/statefulset.yml

```diff
@@ -0,0 +1,108 @@
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: {{ include "pinot.minion.fullname" . }}
+  labels:
+    app: {{ include "pinot.name" . }}
+    chart: {{ include "pinot.chart" . }}
+    component: {{ .Values.minion.name }}
+    release: {{ .Release.Name }}
+    heritage: {{ .Release.Service }}
+spec:
+  selector:
+    matchLabels:
+      app: {{ include "pinot.name" . }}
+      release: {{ .Release.Name }}
+      component: {{ .Values.minion.name }}
+  serviceName: {{ template "pinot.minion.headless" . }}
+  replicas: {{ .Values.minion.replicaCount }}
+  updateStrategy:
+    type: {{ .Values.minion.updateStrategy.type }}
+  podManagementPolicy: Parallel
+  template:
+    metadata:
+      labels:
+        app: {{ include "pinot.name" . }}
+        release: {{ .Release.Name }}
+        component: {{ .Values.minion.name }}
+      annotations:
+{{ toYaml .Values.minion.podAnnotations | indent 8 }}
+    spec:
+      nodeSelector:
+{{ toYaml .Values.minion.nodeSelector | indent 8 }}
+      affinity:
+{{ toYaml .Values.minion.affinity | indent 8 }}
+      tolerations:
+{{ toYaml .Values.minion.tolerations | indent 8 }}
+      containers:
+      - name: minion
+        image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+        imagePullPolicy: {{ .Values.image.pullPolicy }}
+        args: [
+          "StartMinion",
+          "-clusterName", "{{ .Values.cluster.name }}",
+          "-zkAddress", {{ include "zookeeper.url" . | quote }},
+          "-configFileName", "/var/pinot/minion/config/pinot-minion.conf"
+        ]
+        env:
+          - name: JAVA_OPTS
+            value: "{{ .Values.minion.jvmOpts }} -Dlog4j2.configurationFile={{ .Values.minion.log4j2ConfFile }} -Dplugins.dir={{ .Values.minion.pluginsDir }}"
+        ports:
+          - containerPort: {{ .Values.minion.port }}
+            protocol: TCP
+        volumeMounts:
+          - name: config
+            mountPath: /var/pinot/minion/config
+          {{- if .Values.minion.persistence.enabled }}
+          - name: data
+            mountPath: "{{ .Values.minion.persistence.mountPath }}"
+          {{- end }}
+        resources:
+{{ toYaml .Values.minion.resources | indent 12 }}
+      restartPolicy: Always
+      volumes:
+        - name: config
+          configMap:
+            name: {{ include "pinot.minion.config" . }}
+      {{- if not .Values.minion.persistence.enabled }}
+        - name: data
+          emptyDir: {}
+      {{- end }}
+
+  {{- if .Values.minion.persistence.enabled }}
+  volumeClaimTemplates:
+    - metadata:
+        name: data
+      spec:
+        accessModes:
+          - {{ .Values.minion.persistence.accessMode | quote }}
+        {{- if .Values.minion.persistence.storageClass }}
+        {{- if (eq "-" .Values.minion.persistence.storageClass) }}
+        storageClassName: ""
```

Comment:
Some cloud providers will not support a blank storageClassName.

Something like this would be preferable:
```
        {{- if and .Values.server.persistence.storageClass ( not ( empty .Values.server.persistence.storageClass) ) }}
        storageClassName: {{ .Values.server.persistence.storageClass }}
        {{- end }}
```

This also means that the default blank value from`values.yaml` should be removed.

Reply:
Got it, then we should update this for all the components requires pvcs

Reply:
I will update this in a different PR.